src/reducers/ui.js

```diff
@@ -12,15 +12,23 @@ const { prefs } = require("../utils/prefs");
 import type { Action, panelPositionType } from "../actions/types";
 import type { Record } from "../utils/makeRecord";
 
+type searchVisibilityType = {
+    project: boolean,
+    document: boolean
+};
+
 export type UIState = {
-  searchOn: boolean,
+  searchVisibility: Record<searchVisibilityType>,
```

Comment:
Heh, good notes. I had gone with project/document because we had originally called the <kbd>ctrl</kbd>+P search file search but I know we were going to expand it to do more than just search for files. 😄 The nesting can definitely be awkward. I'll adjust it and go for distinct actions for project and file search. 👍